examples/relative_hydration.py

```diff
@@ -0,0 +1,197 @@
+    geometry_engine = FFAllAngleGeometryEngine()
+    proposal_engine = SmallMoleculeSetProposalEngine(
+        [initial_smiles, final_smiles], system_generator, residue_name=mol_name)
+
+    #generate topology proposal
+    topology_proposal = proposal_engine.propose(solvated_system, top_old)
+
+    #generate new positions with geometry engine
+    new_positions, _ = geometry_engine.propose(topology_proposal, pos_old, beta)
+
+    return topology_proposal, pos_old, new_positions
+
+
+def check_alchemical_hybrid_elimination_bar(topology_proposal, old_positions, new_positions, ncmc_nsteps=50, n_iterations=50, NSIGMA_MAX=6.0, geometry=False):
+    """
+    Check that the hybrid topology, where both endpoints are identical, returns a free energy within NSIGMA_MAX of 0.
+    Parameters
+    ----------
+    topology_proposal
+    positions
+    ncmc_nsteps
+    NSIGMA_MAX
+
+    Returns
+    -------
+
+    """
+
+    #make the hybrid topology factory:
+    factory = HybridTopologyFactory(topology_proposal, old_positions, new_positions)
+
+    platform = openmm.Platform.getPlatformByName("CUDA")
+
+    hybrid_system = factory.hybrid_system
+    hybrid_topology = factory.hybrid_topology
+    initial_hybrid_positions = factory.hybrid_positions
+
+
+    #alchemical functions
+    functions = {
+        'lambda_sterics' : '2*lambda * step(0.5 - lambda) + (1.0 - step(0.5 - lambda))',
+        'lambda_electrostatics' : '2*(lambda - 0.5) * step(lambda - 0.5)',
+        'lambda_bonds' : 'lambda',
+        'lambda_angles' : 'lambda',
+        'lambda_torsions' : 'lambda'
+    }
+
+    w_f = np.zeros(n_iterations)
+    w_r = np.zeros(n_iterations)
+
+    #make the alchemical integrators:
+    forward_integrator = NCMCGHMCAlchemicalIntegrator(temperature, hybrid_system, functions, nsteps=ncmc_nsteps, direction='insert')
+    forward_context = openmm.Context(hybrid_system, forward_integrator, platform)
+    print("Minimizing for forward protocol...")
+    forward_context.setPositions(initial_hybrid_positions)
+    for parm in functions.keys():
+        forward_context.setParameter(parm, 0.0)
+
+    openmm.LocalEnergyMinimizer.minimize(forward_context, maxIterations=10)
+
+    initial_state = forward_context.getState(getPositions=True, getEnergy=True)
+    print("The initial energy after minimization is %s" % str(initial_state.getPotentialEnergy()))
+    initial_forward_positions = initial_state.getPositions(asNumpy=True)
+    equil_positions = simulate_hybrid(hybrid_system,functions, 0.0, initial_forward_positions)
+
+    print("Beginning forward protocols")
+    #first, do forward protocol (lambda=0 -> 1)
+    with progressbar.ProgressBar(max_value=n_iterations) as bar:
+        for i in range(n_iterations):
+            equil_positions = simulate_hybrid(hybrid_system, functions, 0.0, equil_positions)
+            forward_context.setPositions(equil_positions)
+            forward_integrator.step(ncmc_nsteps)
```

Comment:
Same as below: You have to call [`forward_integrator.reset()`](https://github.com/choderalab/perses/blob/master/perses/annihilation/ncmc_switching.py#L954-L967) each iteration.